src/backend/api/meals.js

```diff
@@ -1,15 +1,179 @@
+const defaultParams = {
+maxPrice: Number.MAX_VALUE,
+availableReservations: '',
+title: '',
+createdAfter: new Date(2000, 6, 6),
+limit: Number.MAX_VALUE
+};
+
+function unpackParams(request) {
+const sentParams = {};
+
+if (request.query.maxPrice) {
+  const maxPrice = parseInt(request.query.maxPrice);
+  if (isNaN(maxPrice) || maxPrice < 1) {
+    throw new Error('maxPrice param must be a positive number');
+  }
+  sentParams['maxPrice'] = maxPrice;
+}
+
+if (request.query.availableReservations) {
+   const availableReservations = request.query.availableReservations.toLowerCase();
+  if (availableReservations !== 'false' && availableReservations !== 'true' ) {
+    throw new Error('availableReservations param must be a boolean');
+  }
+  sentParams['availableReservations'] = availableReservations;
+}
+
+if (request.query.limit) {
+  const limit = parseInt(request.query.limit);
+  if (isNaN(limit) || limit < 1) {
+    throw new Error('limit param must be a positive number');
+  }
+  sentParams['limit'] = limit;
+}
+
+
+if (request.query.createdAfter) {
+  const createdAfter = Date.parse(request.query.createdAfter);
+  if (isNaN(createdAfter)) {
+    throw new Error('createdAfter param must be a valid date in format YYYY-MM-DD');
+  }
+  sentParams['createdAfter'] = createdAfter;
+}
+
+if (request.query.title) {
+  sentParams['title'] = request.query.title;
+}
+
+return Object.assign({}, defaultParams, sentParams); // Create a new object, then merge defaultParams into that and then sentParams on top of that which overrides the defaults with whatever params were sent and that are valid
+}
+
+
+router.post("/", async (req, res) => {
+try {
+  const mealToAdd={ title: req.body.title, description: req.body.description, location: req.body.location, when: req.body.when, max_reservations: req.body.max_reservations, price: req.body.price, created_date: req.body.created_date }
+  const result = await knex("meal").insert(mealToAdd)
+  res.json(result);
+} catch (error) {
+  throw new Error('something went wrong with adding meal');
+
+}
+});
+
+router.get("/:id", async (req, res) => {
+if (isNaN(parseInt(req.params.id))) {
+  res.status(404).json( {status:'bad request',
+  msq:'ID must be a number'
+  });
+}
+else {
+  try {
+    const mealById = await knex("meal").where({'id': parseInt(req.params.id)});
+    res.send(mealById);
+  } catch (error) {
+    throw new Error('Something went wrong with getting meal by id');
+  }
+}
+});
+
+router.put("/:id", async (req, res) => {
+if (isNaN(Number(req.params.id))) {
+  res.status(404).json({status:'bad request',
+msg:'ID  must be a number'});
+}
+else {
   try {
     // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
+    const titles = await knex("meal").select("title");
```

Comment:
I think this should be named `meals` then you wouldn't have to renamed the dump file above

Reply:
Yes I created a table named meal unintensionally instead of meals .